### ISSUE TITLE
JVM crash on executing win_sparkle_did_find_update callback

### ISSUE THREAD

**Issue by purejava** [purejava/winsparkle-java#1]
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/purejava/winsparkle-java/issues?q=) for duplicates

### Summary

JVM crashes on executing callback

### What software is involved?

- Operating System: Windows 11
- winsparkle-java: 1.0.0

### Steps to Reproduce

1. Define a callback class
```java
public class CB implements win_sparkle_did_find_update_callback_t {
  @Override
  public void apply() {
  }
}
```

2. Set `apply()` method as the callback function to be executed
```java
winsparkle_h.win_sparkle_set_did_find_update_callback(MemoryAllocator.ALLOCATE_CALLBACK_FOR(new CB()));
```

3. Trigger the callback function via WinSparkle

### Expected Behavior

Callback function executes

### Actual Behavior

JVM crashes

### Reproducibility

Always

### Relevant Log Output

```shell
See below
```

### Anything else?

A bug report has been submitted upstream at https://bugreport.java.com/bugreport/crash.jsp

**Comment by purejava** [comment]
[hs_err_pid3060.log](https://github.com/purejava/winsparkle-java/files/10134480/hs_err_pid3060.log)

**Comment by purejava** [comment]
Today Oracle assigned a [bug ID](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8298010) to my bug report.

**Comment by purejava** [comment]
Oracle looked into the report and created [an additional bug report](https://bugs.openjdk.org/browse/JDK-8298010), that describes the root cause for this issue and suggests a solution:

**Root cause**
The callback is eligible for garbage collection after the setter returns, since there are no strong references to it. It is possible (or likely) that the callback is freed before it is called, and the call will jump through a dangling pointer instead, which would likely cause a crash.

**Solution**
Replace the use of an implicit memory session with a shared session and allocate the memory for a callback (aka setting the pointer to a function that is executed as the callback) in the shared session.

**Comment by purejava** [comment]
Fixed by 66c49e1a18cc5846bfb4ffc0617d30d16b8726b7 and released with [version 1.1.0](https://github.com/purejava/winsparkle-java/releases/tag/1.1.0).
